### PR TITLE
[BugFix] Fix can't show tables when db contains both hive and iceberg tables

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fix can't show tables when db contains both hive and iceberg tables.
Show the below error:
```sql
mysql> show tables;
ERROR 1064 (HY000): Not an iceberg table: iceberg_test.db.table_migration_par (type=null)
```

Now in the iceberg catalog, we will only display the iceberg table.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java|0|6|00.00%|[915, 921, 922, 923, 924, 925]|

File: fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 1]
```diff
@@ -1011,13 +912,17 @@ public Database getDatabase(String catalogName, String databaseName)
...
 
     @Override
     public List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
             throws MetaException, InvalidOperationException, UnknownDBException, TException {
-        throw new TException("method not implemented");
+        GetTablesRequest req = new GetTablesRequest(dbName);
+        req.setCatName(catName);
+        req.setTblNames(tableNames);
+        req.setCapabilities(version);
+        return client.get_table_objects_by_name_req(req).getTables();
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:925]: Is it interface compatible with older HMS versions?

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:925]: I'm not sure, I haven't an older hms version to have a test.

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:925]: maybe you need to test if you can access 1.x/2.x/3.x hms

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java]: I am using lower API instead, it already has in hive-1.xxx.

https://github.com/apache/hive/blob/branch-1/metastore/if/hive_metastore.thrift

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java|0|2|00.00%|[1014, 1020]|

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.1

**Pull request merged**